src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/PackageRestore/PackageRestoreDataSource.cs

```diff
@@ -275,12 +240,78 @@ private RestoreData CreateRestoreData(ProjectRestoreInfo restoreInfo, bool succe
                 succeeded: succeeded && lastWriteTime != DateTime.MinValue);
         }
 
+        protected virtual bool IsProjectConfigurationVersionOutOfDate(IReadOnlyCollection<PackageRestoreConfiguredInput> configuredInputs)
```

Comment:
nit: The logic for this method is very similar to the `IsSavedNominationOutOfDate` method I commented on above. Any chance for consolidation would be welcomed.